File: pgm/ABC_Dart/ABC_376_A.dart
// import 'dart:collection';
import 'dart:io';
// import 'dart:math';

void main() {
  final [_, C] = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  final T = stdin.readLineSync()!.split(' ').map(int.parse).toList();
  const int one = 1;
  const int zero = 0;
  int pre = -C;
  int Candy = zero;
  for (int time in T) {
    if (time - pre >= C) {
      Candy += one;
      pre = time;
    }
  }
  print(Candy);
  exit(0);
}

File: pgm/ABC_Kotlin/ABC_376_A.kt
// import java.time.*

// import kotlin.math.*

const val one: Int = 1
const val zero: Int = 0
const val space: String = " "

fun main() {
    val (N: Int, C: Int) = readln().split(space).map { it.toInt() }
    val T: List<Int> = readln().split(space).map { it.toInt() }
    var pre:Int = -C
    var Candy:Int = zero
    for (time in T) {
        if (time - pre >= C) {
            Candy += one
            pre = time
        }
    }
    println(Candy)
    kotlin.system.exitProcess(0)
}